process-phoenix/src/main/java/com/jakewharton/processphoenix/ProcessPhoenix.java

```diff
@@ -53,25 +55,83 @@ public static void triggerRebirth(Context context) {
     triggerRebirth(context, getRestartIntent(context));
   }
 
+  /**
+   * Call to restart the application process using the provided targetClass as an intent.
+   * <p>
+   * Behavior of the current process after invoking this method is undefined.
+   */
+  public static void triggerRebirth(Context context, Class<?> targetClass) {
```

Comment:
I don't really like this from an API design standpoint. I cannot imagine the case where you have either an activity or a service and need to call some general-purpose method to recreate it. Chances are you know what you have at the call-site. Someone that actually doesn't know can write the assignable check themselves.

If we want to offer `Class`-accepting methods then we can overload the specific methods with `Class<? extends Activity>` and `Class<? extends Service>`.

Reply:
Ahaha ouch. Bad 7-year-ago Jake!